python/lsst/ap/pipe/metrics.py

```diff
@@ -187,3 +189,133 @@ def run(self, matchedFakes, band):
         meas = Measurement(self.config.metricName,
                            len(magCutFakes) * u.count)
         return Struct(measurement=meas)
+
+
+class PipelineTimingMetricConnections(
+        MetricTask.ConfigClass.ConnectionsClass,
+        dimensions={"instrument", "visit", "detector"},
+        defaultTemplates={"labelStart": "",
+                          "labelEnd": "",
+                          "package": "ap_pipe",
+                          "metric": "ApPipelineTime"}):
+    metadataStart = connTypes.Input(
+        name="{labelStart}_metadata",
+        doc="The starting task's metadata.",
+        storageClass="TaskMetadata",
+        dimensions={"instrument", "exposure", "detector"},
+        multiple=False,
+    )
+    metadataEnd = connTypes.Input(
+        name="{labelEnd}_metadata",
+        doc="The final task's metadata.",
+        storageClass="TaskMetadata",
+        dimensions={"instrument", "visit", "detector"},
+        multiple=False,
+    )
+
+
+class PipelineTimingMetricConfig(MetricTask.ConfigClass, pipelineConnections=PipelineTimingMetricConnections):
+    # Don't include the dimensions hack that MetadataMetricConfig has; unlike
+    # TimingMetricTask, this task is not designed to be run on multiple
+    # pipelines.
+    targetStart = pexConfig.Field(
+        dtype=str,
+        doc="The method to profile on the starting task, optionally prefixed by "
+            "one or more tasks in the format of "
+            "`lsst.pipe.base.Task.getFullMetadata()`.")
```

Comment:
Maybe it's because I'm not a native English speaker, but I can't parse the proposed replacement.

"The method to take as the starting point on the starting task"? Ignoring the redundant redundancy...

Reply:
Oh, yes, that's actually much more clear.

Reply:
Also replaced "one or more tasks" with "one or more task names", hopefully that reduces confusion.